packages/server-wallet/src/wallet/wallet.ts

```diff
@@ -288,6 +288,38 @@ export class SingleThreadedWallet
     return response.multipleChannelOutput();
   }
 
+  /**
+   * Trigger a response containing a message for counterparties, with all objectives and channels for a set of objectives.
+   * @param objectiveIds The ids of the objectives that should be sent to the counterparties
+   * @returns A promise that resolves to an object containing the messages.
+   */
+  public async syncObjectives(objectiveIds: string[]): Promise<MultipleChannelOutput> {
+    const response = WalletResponse.initialize();
+    const objectives = await this.store.getObjectivesByIds(objectiveIds);
+
+    const fetchedObjectiveIds = objectives.map(o => o.objectiveId);
+    const missingObjectives = _.difference(objectiveIds, fetchedObjectiveIds);
+
+    if (missingObjectives.length > 0) {
+      this.logger.error(
+        {objectiveIds: missingObjectives},
+        'The following objectives cannot be found'
+      );
+      throw new Error('Could not find all objectives');
+    }
+
+    for (const o of objectives) {
+      const channel = await this.store.getChannelState(o.data.targetChannelId);
```

Comment:
Do we really need to fetch the `channel` twice? (`_syncChannel` also fetches the channel.)

When I look at the implementation of `_syncChannel`, it appears difficult to pull this off.

Reply:
I think we could do this with some refactoring but I'm not sure it's worth the effort right now?

Reply:
Agreed.

I think it's worth marking this, either with a comment or an issue. I'll leave that up to you.

Reply:
Added a comment in https://github.com/statechannels/statechannels/pull/3275/commits/8158ea764c92be74378850d52591b4d28fb627df